client/my-sites/controller.js

```diff
@@ -235,14 +241,19 @@ module.exports = {
 	 * Set up site selection based on last URL param and/or handle no-sites error cases
 	 */
 	siteSelection( context, next ) {
+		const { state, dispatch } = getStore( context );
+
 		const siteID = route.getSiteFragment( context.path );
 		const basePath = route.sectionify( context.path );
 		const currentUser = user.get();
 		const hasOneSite = currentUser.visible_site_count === 1;
 		const allSitesPath = route.sectionify( context.path );
+		const primaryId = getPrimarySiteId( state );
+		const primary = getSite( state, primaryId ) || '';
+		const hasInitialized = !! getSites( state );
```

Comment:
Is the intention here to wait for fresh data, or is persisted data fine? This will be true if we have some sites in indexedDB.